lib/extensions.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+const path = require('path');
+
+module.exports = {
+
+  withConflictExclusions: function(BaseGenerator) {
+    return class extends BaseGenerator {
+      constructor(args, opts) {
+        super(args, opts);
+
+        this._conflictExclusions = [];
```

Comment:
Please use map rather than array, that will prevent duplicates and will be more efficient